plugins/DICOMwebServer.js

```diff
@@ -895,26 +895,42 @@ async function dicomwebserver(fastify) {
             ],
           }),
           (result) => {
-            try {
-              const jsonResult = JSON.parse(result);
-              const map = {};
-              const res = jsonResult.container ? JSON.parse(jsonResult.container) : [];
-              res.forEach((item) => {
-                if (item['0020000E'])
-                  map[item['0020000E'].Value[0]] = item['0008103E']
-                    ? item['0008103E'].Value[0]
-                    : '';
-              });
-              resolve({ data: res });
-            } catch (err) {
-              console.log(err);
-              reject(err);
-            }
+            resolve(result);
           }
         );
       })
   );
 
+  fastify.decorate(
+    'getStudySeriesDIMSE',
+    (studyUID) =>
+      new Promise((resolve, reject) => {
+        const dimsePromises = [
+          fastify.promisifyDIMSE(config.dimse, studyUID),
+          fastify.promisifyDIMSE(config.vnaDimse, studyUID),
+        ];
+        Promise.all(dimsePromises).then((results) => {
+          try {
+            // use vna if there is a successfull result from vna
+            // it means the study is already archived
+            // we assume the series data does not change one it is archived
+            const resultsJSON = results.map((item) => JSON.parse(item));
+            const jsonResult =
+              resultsJSON[1].code === 0 && resultsJSON[1].container ? resultsJSON[1] : results[0];
+            const map = {};
+            const res = jsonResult.container ? JSON.parse(jsonResult.container) : [];
+            res.forEach((item) => {
+              if (item['0020000E'])
+                map[item['0020000E'].Value[0]] = item['0008103E'] ? item['0008103E'].Value[0] : '';
```

Comment:
Where do we use this map object?

Reply:
good catch! apparently, I created it for logging, I deleted the console.log but not the map